### ISSUE TITLE
aria-label in Side Nav Mobile causes repeat of label when navigating to content

### ISSUE THREAD

**Issue by dv297** [department-of-veterans-affairs/va.gov-team#131854]
### Description

The Side Navigation component utilizes the VADS Side Navigation component, which under the hood utilizes the USWDS Accordion component. 

Based on the current implementation, when a veteran navigates the Side Nav using VoiceOver, they get these prompts:

1. "heading level 2, Related Pages Menu, expanded, button"
2. "Related Pages Menu, Navigation"
3. "List 3 items"

This is redundant information that can confuse the veteran on why their navigation may not have progressed. This is likely caused by using aria-label rather than aria-labelledby

### How severe is the problem?

- [ ] Critical - A Veteran is blocked or significantly hindered from submitting the form, data is lost, or claim and evidence test coverage is missing. Fix ASAP.
- [x] Moderate - A Veteran is not blocked, but the happy path is interrupted. Fix after critical bugs.
- [ ] Minor - Cosmetic bugs and minor language inconsistencies. Defer to sprint planning.

### Desired behavior

The screen reader should be able to connect that the accordion button is related to the content of the accordion.

https://html.spec.whatwg.org/multipage/sections.html#the-aside-element

### Tasks

- Update the Side Nav button so that it and the content of the accordion can be related using aria-labelledby

### Acceptance Criteria

- [ ] Desired behavior is met
- [ ] Desired behavior is validated using screen reader
- [ ] Code reviewed and approved
- [ ] Bugfix tested and reviewed by Design

### Definition of Done

- [ ] Meets acceptance criteria
- [ ] Reviewed and approved by product and/or design

### Engineering

- [ ] All tests pass
- [ ] New functionality is covered by unit tests
- [ ] Logging and monitoring are implemented (if applicable)

### Code Review & Pull Requests

- [ ] PR includes Local testing steps
- [ ] PR includes Flipper/testing state details (if applicable)
- [ ] PR includes Author's local proof of submission screenshot
- [ ] Copilot review completed and feedback addressed
- [ ] Internal reviewer approved
- [ ] Internal reviewer added local proof of submission screenshot
- [ ] Code functionality verified on Staging after merge

### Refinement Checklist!

- [ ] Added description, tasks, and acceptance criteria
- [ ] Added estimate
- [ ] Labeled with Practice Area (engineer, design, product, data science)
- [ ] Labeled with issue type and characteristics of the ticket (bug, accessibility, request, discovery, documentation, research, content, ux testing, front-end, back-end, datadog, etc.)
- [ ] Added any other relevant project fields (team, OCTO priority...)
- [ ] Added an Epic or Super Epic
